crates/sdk/src/prover.rs

```diff
@@ -109,8 +113,12 @@ pub(crate) fn verify_proof<C: SP1ProverComponents>(
                 .collect_vec();
 
             // Make sure the committed value digest matches the public values hash.
-            for (a, b) in committed_value_digest_bytes.iter().zip_eq(bundle.public_values.hash()) {
-                if *a != b {
+            // It is computationally infeasible to find two distinct inputs, one processed with
+            // SHA256 and the other with Blake3, that yield the same hash value.
+            for (a, (sha_b, blake3_b)) in committed_value_digest_bytes.iter().zip_eq(
+                bundle.public_values.hash().into_iter().zip(bundle.public_values.blake3_hash()),
+            ) {
+                if *a != sha_b && *a != blake3_b {
                     return Err(SP1VerificationError::InvalidPublicValues);
                 }
             }
```

Comment:
This code seems overcomplicated. We can simplify by just checking whether the `public_values_hash` of either `blake3` or `sha2` matched the `committed_value_digest_bytes`?